### PR TITLE
Update chpldoc dependencies

### PR DESCRIPTION
**Pull request by mppf**
This PR updates the chpldoc requirements to the newest versions. I tried this because I was thinking about adjustment to the search box, and I wanted to see if the newer versions of Sphinx behave differently. (They do not seem to).

Sphinx 7.2.6 requires Python >= 3.9, so this PR updates prereqs.rst to list 3.9 as required for chpldoc, start_test, and c2chapel. (The previous requirement for these was Python 3.7).

It adds the sphinx extension `sphinxcontrib.jquery` to `conf.py` because the search box from sphinx-rtd-theme now uses jquery.

| package | version before this PR | version after this PR |
| ------- | ---------------------- | --------------------- |
| MarkupSafe | 2.1.3 | 2.1.5 |
| Sphinx | 5.3.0 | 7.2.6 |
| urllib3 | 2.1.0 | 2.2.1 |
| docutils | 0.18 | 0.20.1 |
| sphinxcontrib-chapeldomain | 0.0.31 | 0.0.32 |

(note that https://github.com/chapel-lang/sphinxcontrib-chapeldomain/pull/92 updated sphinxcontrib-chapeldomain in a coordinated manner: docutils 0.18 -> 0.20.1 and Sphinx 5.3.0 -> 7.2.6. The result of that PR is available in sphinxcontrib-chapeldomain  version 0.0.32.)

Reviewed by @lydia-duncan - thanks!

- [x] full comm=none testing

### CODE REVIEW

File: doc/rst/conf.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
 # ones.
 extensions = [
     'sphinx.ext.todo',
+    'sphinxcontrib.jquery',
```
Comment (mppf) [inline doc/rst/conf.py:36]: the search box from sphinx-rtd-theme now uses jquery

File: third-party/chpl-venv/chpldoc-requirements3.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # Split into 3 files to work around problems with CHPL_PIP_FROM_SOURCE
 sphinx-rtd-theme==2.0.0
-sphinxcontrib-chapeldomain==0.0.31
+./sphinxcontrib-chapeldomain
```
Comment (mppf) [inline third-party/chpl-venv/chpldoc-requirements3.txt:3]: This needs to be adjusted once the sphinxcontrib-chapeldomain has its patch

File: frontend/doc/doxygen-pitfalls.txt [hunk 3]
```diff
@@ -12,6 +12,7 @@ Doxygen will fail to resolve:
 
   - C++ standard library types (that are not builtin, e.g. `int64_t`)
   - Any `compiler/dyno` types that are generated by macros (for now)
```
Comment (lydia-duncan) [inline frontend/doc/doxygen-pitfalls.txt:14]: Can we remove this line (or should we update it to say we think it should be resolved now)?

File: third-party/chpl-venv/chpldoc-requirements1.txt [hunk 4]
```diff
@@ -1,2 +1,2 @@
 # Split into 3 files to work around problems with CHPL_PIP_FROM_SOURCE
-MarkupSafe==2.1.3
+MarkupSafe==2.1.5
```
Comment (lydia-duncan) [inline third-party/chpl-venv/chpldoc-requirements1.txt:2]: We've typically noted such updates in our CHANGES file (either generally or individually), so it'd probably be good to either update your PR title to indicate you're modifying more than just Sphinx or list the version updates in your PR message

**Review by lydia-duncan — APPROVED** [review]
Really really minor suggestions, thanks for running with this!

File: frontend/doc/doxygen-pitfalls.txt (hunk 3 above)
Comment (mppf) [reply frontend/doc/doxygen-pitfalls.txt:14]: This line is still accurate and the problem has not been resolved, AFAIK. We use workarounds.

Comment (lydia-duncan) [reply frontend/doc/doxygen-pitfalls.txt:14]: Ah, okay.  Thanks!

**Pull request merged**